### PR TITLE
Option to add additional policies to the worker iam role

### PR DESCRIPTION
**Pull request by francishokh**
It can be used to attach existing ecr pull policy.

### CODE REVIEW

**Review by niroz89 — COMMENTED** [review]
may be we can extend https://github.com/SPHTech-Platform/terraform-aws-eks/blob/630f5ddec41e4b5eb471dd02b9af0adf7b07520d/iam.tf#L52 itself to support all additional policies ?

similar to what's done in upstream module  as below.
https://github.com/terraform-aws-modules/terraform-aws-eks/blob/62dddaedb997758348f9c0324a170c0e11e1d692/main.tf#L289

@lawliet89 what do you think ?

File: modules/essentials/ecr.tf [hunk 1]
```diff
@@ -36,3 +36,10 @@ resource "aws_iam_role_policy_attachment" "worker_ecr_pullthrough" {
   role       = var.worker_iam_role_name
   policy_arn = aws_iam_policy.ecr_cache[0].arn
 }
+
+resource "aws_iam_role_policy_attachment" "worker_ecr_existing_pullthrough" {
+  count = !var.configure_ecr_pull_through && var.attach_existing_ecr_policy ? 1 : 0
```
Comment (lawliet89) [inline modules/essentials/ecr.tf:41]: Why not only attach when `var.existing_ecr_policy_arn` is not empty.

```suggestion
  count = !var.configure_ecr_pull_through && var.existing_ecr_policy_arn != "" && var.existing_ecr_policy_arn != null ? 1 : 0
```

Can remove one variable.

**Comment by lawliet89** [comment]
Good idea. Let's do that.

**Review by niroz89 — COMMENTED** [review]
remove commented items, generate README.md

**Comment by francishokh** [comment]
Changes have been done, with help from Uchinda
@niro, for your review

**Review by niroz89 — APPROVED** [review]
LGTM

**Pull request merged**